test/fixtures/dialects/exasol/SelectStatement.yml

```diff
@@ -437,48 +434,45 @@ file:
           start_bracket: (
           values_clause:
           - keyword: VALUES
-          - bracketed:
-            - start_bracket: (
-            - expression:
-                literal: "'a'"
-            - comma: ','
-            - expression:
-                literal: "'b'"
-            - comma: ','
-            - expression:
-                literal: "'c'"
-            - end_bracket: )
+          - expression:
```

Comment:
It's interesting how we're adding `expression` around some things and removing it from others. Why's this happening? Any concerns?

Reply:
Basically, a lot are being removed as expression is a bit of a catch-all. I've adding that trimming into the longest_match comparison, and the concept of a terminator, so it's prefering some simpler types. It is adding some expressions, but these match their analogues in ANSI. so I think it's simply reordering that will remove them. Either way, not too worried, as it didn't cause any issues in tests.

Reply:
Thanks, I was just curious.

Reply:
I did a bit of work in that area recently -- requiring that if a subdialect replaces a parent dialect's segment definition, the type must remain the same. It would be good to keep improving in this area. Automated internal quality checks are great at preventing bugs. 🙌

Reply:
@barrywhart - could you make an issue describing exactly what you mean here? I think it sounds really interesting and you've clearly thought about it more than I have :). Feel free to assign it to me, I've been doing quite a lot of looking at common structures recently.

Reply:
@alanmcruickshank: I'll create an issue. At the moment, I don't have any specific ideas for additional checks. Mostly wanted to connect your ideas to that earlier work, as something to possibly align with (e.g. where to put the code, how to report errors, etc. etc.).

The recent work I mention was in `src/sqlfluff/core/dialects/base.py` in #2942.

Reply:
@alanmcruickshank: I created issue #2977.